src/test/battle/battle.test.ts

```diff
@@ -312,5 +327,31 @@ describe("Test Battle Phase", () => {
     await game.toNextWave();
     expect(game.scene.currentBattle.waveIndex).toBeGreaterThan(waveIndex);
   }, 20000);
+
+  it("does not force switch if active pokemon faints at same time as enemy mon and is revived in post-battle", async () => {
+    const moveToUse = Moves.TAKE_DOWN;
+    game.override
+      .battleType("single")
+      .starterSpecies(Species.SAWK)
+      .enemySpecies(Species.RATTATA)
+      .startingWave(1)
+      .startingLevel(100)
+      .moveset([moveToUse])
+      .enemyMoveset(new Array(4).fill(Moves.SPLASH))
```

Comment:
same with other tests
```suggestion
      .enemyMoveset(SPLASH_ONLY)
```